CWE-94-Python_darwin/GAIR-NLP_PC-Agent/fs.py

```
@@ -0,0 +1,78 @@
+
+    def reset(self):
+        if self.backup_path:
+            # Ensure the folder exists
+            ensure_folder(self.folder_path)
+
+            # Attempt to clear the original folder
+            MAX_RETRIES = 2
+            for attempt in range(MAX_RETRIES):
+                try:
+                    # On Windows, use system command to delete
+                    subprocess.call(f'rd /s /q "{self.folder_path}"', shell=True)
```

Ruff (0.13.2)

`subprocess` call with `shell=True` identified, security issue

(S602)